### ISSUE TITLE
Evaluation of well position effect

### ISSUE THREAD

**Issue by jessica-ewald** [broadinstitute/2021_09_01_VarChAMP#20]
@Zitong-Chen-16 @shntnu @AnneCarpenter : the concerning well position analysis results. Would love to hear your thoughts!

### Background
The team generated a few batches that contain many repeated controls to try understand well position effects. There are two batches, each with four plates. Each plate contains 48 wells with ALK_REF, and 48 well with ALK_VAR (referred to as "REF" and "VAR" from here on). The 'ALK' locus was chosen because it is a positive protein mis-localization control. 

The objective here was to determine how well our production data analysis strategy detects phenotypic effects (morphological/localization differences between REF and VAR) compared to technical well position effects. 

Our current method for calling variants with significant phenotypic effects is to train a binary classifier to distinguish a given VAR from its WT using three plates within a batch, and to test on the fourth plate. We measure success using an F1 score. Since we expect there to be some well position effect, we also train binary classifiers to differentiate cells from pairs of repeated control wells (the well position null). The REF-VAR F1 scores are compared to the distribution of control F1 scores to decide which pairs of REF-VAR wells outperform the well position null. This well position null is imperfect, because repeated controls are usually closer together than REF-VAR pairs (need to confirm this). Therefore, we did this larger analysis described here. 

### Methods
Features were split into 3 sets: protein features, non-protein fluorescent features (3 channels), and brightfield features (3 channels). Considering the 48 REF wells and 48 VAR wells, we define every possible pair of two well positions: REF-REF pairs (48 choose 2 = 1128), VAR-VAR pairs (48 choose 2 = 1128), and REF-VAR pairs (48*48 = 2304). 

For each feature set (protein, non-protein, brightfield) and comparison (REF-REF, VAR-VAR, REF-VAR) we train a binary XGBoost classifier with three plates from one batch and test with the fourth plate from that batch. We also train one null for reference - the REF-REF pairs are used to train another classifier after shuffling the well position labels of the training data. The F1 score and feature importance scores are saved from each classifier. 

Since this is all quite computationally intensive, this analysis considers only the four plates in batch 4 (ignoring batch 6) and only looks at one training & test set (there are 4 possible sets, one with each of the 4 plates as the test plate).

### Results

**Figure 1:** the distribution of F1 scores across all classifiers, faceted by comparison type and feature set.

![image](https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/4fa36a10-ca30-4c8a-9680-efc9c2d36b58)

The takeaways here:
- The distribution of F1 scores are virtually identical between REF-VAR, REF-REF, and VAR-VAR pairs
- The brightfield features are best at predicting well position, next best are non-protein features, and next protein features
- The null is centered on 0.5 and has a relatively narrow spread. For both the protein and non-protein fluorescent channels, there are classifiers that do significantly better and significantly worse than the NULL. Sam and I have talked through a potential explanation for this.
- The distributions of protein and non-protein F1 scores unfortunately look suspiciously similar to the distribution of F1 scores in our production classifiers from this [notebook](https://github.com/broadinstitute/2021_09_01_VarChAMP/blob/main/6.downstream_analysis/notebooks/1.phenotype_analysis_final.ipynb) (comparing all REF-VAR pairs across all loci in our real data) - look at the orange 'allele' distributions and compare to the protein and non-protein distribution shapes above:
<img width="574" alt="image" src="https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/cf12cdc4-ee44-4046-9773-6b6ffe2fbff2">
<img width="584" alt="image" src="https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/6280c1fc-dff2-4541-8500-f8b46ff41955">

**Figure 2:** a comparison of feature importance scores across comparison types. Feature importance (FIP) scores were ranked for each classifier (1 = best, highest = worst) and averaged across all classifiers with the same comparison type and feature set. Here are pairwise scatterplots of mean FIP scores from the protein feature classifiers for all comparison types. Each point corresponds to a single feature.
![image](https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/a8c195b5-a2be-4b54-8461-267968d23f89)

Key takeaways:
- The classifiers are learning to use the same features regardless of whether we are looking at REF-REF, VAR-VAR, or REF-VAR pairs of wells. We appear to always be learning well position and not morphology (😱) 
- The null prioritizes random features (as expected from a shuffled null)
- These takeways are the same for the non-protein and brightfield classifiers

**Figure 3:** a scatterplot comparing the F1 score for the classifier for each pair of wells to the physical distance between those two wells on the plate. 
![image](https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/4de5af45-0422-4c86-9e79-b3ebfb70fee4)

Key takeways:
- Only the protein feature REF-REF, VAR-VAR, and REF-VAR classifiers have a significant correlation between F1 score and physical distance (corr r = 0.16, 0.22, 0.24, corr p-value = 10-8, 10-14, 10-35 respectively).

**Comment by jessica-ewald** [comment]
Updating this for our records: this is solved for the protein channel ('localization') but not for the non-protein channel ('morphology'). I will open a new issue to discuss the non-protein classifiers. 

The key step was removing the well position effect correction in the pre-processing pipeline. We ported this over from the JUMP pipeline where the mean of each feature for each well position is subtracted from the data. For JUMP this is fine because there are many, many platemaps and so if there is any well position pattern, we are highly confident that it is technical. However, for VarChamp there are only ~5 platemaps and so by deleting the well position pattern, we are actually deleting a ton of the phenotypic signal. Removing this step and reprocessing the control scramble data shows separation between the WT-VAR and CTRL-CTRL distributions for the protein data for both the ALK WT-VAR and LPAR1-MAPK9 comparisons:

**LPAR1-MAPK9 - protein channel**
![image](https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/da9183ca-ccc2-4184-8aff-20a6f55cc764)

**ALK WT-VAR - protein channel**
![image](https://github.com/broadinstitute/2021_09_01_VarChAMP/assets/33487514/90a95a4b-dbbe-4c67-b9e7-cd424e30832c)

Notebook with these plots is [here](https://github.com/broadinstitute/2021_09_01_VarChAMP/blob/main/7.downstream_analysis_jess/notebooks/4F_well_position_scen4_mapk_lpar1.ipynb).
